src/placement/pl_map_common.c

```diff
@@ -63,6 +63,50 @@ remap_add_one(d_list_t *remap_list, struct failed_shard *f_new)
 	d_list_add(&f_new->fs_list, remap_list);
 }
 
+void
+reint_add_one(d_list_t *remap_list, struct failed_shard *f_new)
+{
+	struct failed_shard     *f_shard;
+	d_list_t                *tmp;
+
+	D_DEBUG(DB_PL, "Reint target : fnew: %u", f_new->fs_shard_idx);
+
+	/* All failed shards are sorted by fseq in ascending order */
+	d_list_for_each_prev(tmp, remap_list) {
+		f_shard = d_list_entry(tmp, struct failed_shard, fs_list);
+
+		D_DEBUG(DB_PL, "fnew: %u, fshard: %u", f_new->fs_shard_idx,
+			f_shard->fs_shard_idx);
+
+		if (f_new->fs_shard_idx < f_shard->fs_shard_idx)
+			continue;
+		if (f_new->fs_shard_idx == f_shard->fs_shard_idx)
+			return;
```

Comment:
what's the purpose of keeping the list in shard_idx order?

Reply:
See above.